### DISCUSSION TITLE
Ad-hoc query to get mean run duration of dag

### DISCUSSION THREAD

**Discussion by hurcy** [apache/airflow#29955] [Q&A]
Hi,

I'm using airflow 2.4.2.

I would like to get mean run duration of all dags using ad-hoc query in postgresql.

<img width="683" alt="image" src="https://user-images.githubusercontent.com/982262/223445761-aa7c4431-9e6f-4b7d-be51-44c0338e8c37.png">

I tried this query, but it seems not giving me correct duration.
because it took much longer than the calculated duration.

```
select 
  	dag_id, dag_hash, execution_date,  round(extract(epoch from avg(end_date - start_date))::numeric,2) as  duration
  from dag_run
  where state = 'success'
  group by dag_id, dag_hash, execution_date
  order by execution_date desc;
```

**Comment by potiuk** [comment]
Is this a question? What do you mean by "it took longer?" What did you expect to see ? Did you try to use statsd metrics produced by Airflow instead of querying the database? (Database of airflow is an internal implementation detail and might change any time).

**Reply by hurcy** [reply]
> Is this a question? What do you mean by "it took longer?" What did you expect to see ? Did you try to use statsd metrics produced by Airflow instead of querying the database? (Database of airflow is an internal implementation detail and might change any time).

Thanks for the info. I'll try statsd metrics!
